### ISSUE TITLE
spice-client-glib-usb-helper: Error setting facl: Operation not permitted

### ISSUE THREAD

**Issue by Enteee** [NixOS/nixpkgs#103981]
**Describe the bug**
After merging: https://github.com/NixOS/nixpkgs/pull/97826 spice usb redirection in virt-viewer still yields: `spice-client-glib-usb-helper: Error setting facl: Operation not permitted`

**To Reproduce**
1. set: `virtualisation.spiceUSBRedirection.enable = true`
2. Start virt-viewer from command line and attach to vm
3. Redirect a usb device to vm

**Expected behavior**
USB device should be redirected to VM.

**Additional context**
According to the documentation [here](https://github.com/NixOS/nixpkgs/blob/002a5f9de16130ac9f8e474de75d9b5eca7bf4bd/nixos/modules/virtualisation/spice-usb-redirection.nix#L7), `spice-client-glib-usb-acl-helper` should be installed with setuid permissions. But on my system it is not. `sudo chmod u+s $(which spice-client-glib-usb-acl-helper)` is a workaround for the issue.

Do we maybe need a `setuid = true;` around [here](https://github.com/NixOS/nixpkgs/blob/002a5f9de16130ac9f8e474de75d9b5eca7bf4bd/nixos/modules/virtualisation/spice-usb-redirection.nix#L19) ?

**Notify maintainers**
* @lheckemann 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.77, NixOS, 21.03pre252083.2deeb58f494 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.8`
 - channels(root): `"home-manager, nixos-21.03pre252083.2deeb58f494, nixos-hardware"`
 - channels(ente): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
  - virtualisation.spiceUSBRedirection
```

**Comment by lheckemann** [comment]
No, it shouldn't need setuid, the fowner capability [as set by the module](https://github.com/NixOS/nixpkgs/pull/97826/files#diff-47f4417b57b1d8a86d2588e3101694c353969055d71a7c3d1415f7c36261bc87R19) should be enough… Is it set correctly on your system? Check using `getcap /run/wrappers/bin/spice-client-glib-usb-acl-helper`.

**Comment by Enteee** [comment]
```
$ getcap /run/wrappers/bin/spice-client-glib-usb-acl-helper
/run/wrappers/bin/spice-client-glib-usb-acl-helper cap_fowner,cap_setpcap=ep
```

**Comment by tolbrino** [comment]
I ran into the same issue. The aforementioned `setuid` workaround did work as well.

**Comment by marcus7070** [comment]
I can reproduce this. `getcap` returned exactly the same as @Enteee. Only after applying workaround does usb redirection work.

**Comment by Artturin** [comment]
tested in virt-manager and it works

**Comment by jtojnar** [comment]
We might have actually switched to capabilities too soon. The MR implementing them has only been merged in 0.39, see https://github.com/NixOS/nixpkgs/pull/152044/files.
